watertap/flowsheets/dye_desalination/dye_desalination.py

```diff
@@ -716,27 +725,51 @@ def add_costing(m):
             ),
             doc="Specific energy consumption of the treatment train on a feed flowrate basis [kWh/m3]",
         )
+    elif hasattr(m.fs, "dewaterer") or hasattr(m.fs, "gac"):
+        m.fs.ro_costing.cost_process()
+
+        m.fs.specific_energy_intensity = Expression(
+            expr=m.fs.zo_costing.electricity_intensity,
+            doc="Specific energy consumption of the treatment train on a feed flowrate basis [kWh/m3]",
+        )
     else:
         m.fs.specific_energy_intensity = Expression(
-            expr=(m.fs.zo_costing.electricity_intensity),
+            expr=m.fs.zo_costing.electricity_intensity,
             doc="Specific energy consumption of the treatment train on a feed flowrate basis [kWh/m3]",
         )
 
     # Annual disposal of brine
     if hasattr(m.fs, "desalination"):
-        m.fs.brine_cost = Expression(
-            expr=(
-                m.fs.zo_costing.utilization_factor
-                * (
-                    m.fs.zo_costing.waste_disposal_cost
-                    * pyunits.convert(
-                        m.fs.brine.properties[0].flow_vol,
-                        to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+        if brine_revenue:
+            m.fs.brine_cost = Expression(
+                expr=(
+                    -1
+                    * m.fs.zo_costing.utilization_factor
```

Comment:
Is the zero order costing utilization factor same as ro.costing? Can you tell me some background why we need to use zero order costing in this flowsheet?

Reply:
Yes, both utilization factors are set to 0.85. Zero order costing is needed to cost the ZO nanofiltration and pump models The `dye_desalination_global_costing.yaml` file is also linked to the zero order costing and provides values like `dye_disposal_cost`, `recovered_water_cost`, `electricity_cost`, etc.